test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,223 @@
+        const std::string big_cache_size = "cache_size=500MB";
+
+        while (tc->running()) {
+            tc->sleep();
+
+            /* Get the current cache size. */
+            uint64_t prev_cache_size = conn_impl->cache_size;
+
+            /* Reconfigure with the new cache size. */
+            testutil_check(conn->reconfigure(
+              conn, increase_cache ? big_cache_size.c_str() : small_cache_size.c_str()));
+
+            /* Get the new cache size. */
+            uint64_t new_cache_size = conn_impl->cache_size;
+
+            logger::log_msg(LOG_TRACE,
+              "The cache size was updated from " + std::to_string(prev_cache_size) + " to " +
+                std::to_string(new_cache_size));
+
+            /*
+             * The collection id and the key are dummy fields which are required by the
+             * save_operation API but not needed for this test.
+             */
+            const uint64_t collection_id = 0;
+            const std::string key, value = std::to_string(new_cache_size);
+
+            /* Retrieve the current transaction id. */
+            uint64_t txn_id = ((WT_SESSION_IMPL *)tc->session.get())->txn->id;
+
+            /* Save the change of cache size in the tracking table. */
+            tc->transaction.begin();
+            int ret = tc->tracking->save_operation(txn_id, tracking_operation::CUSTOM,
+              collection_id, key, value, tc->tsm->get_next_ts(), tc->op_track_cursor);
+
+            if (ret == 0)
+                testutil_assert(tc->transaction.commit());
+            else {
+                /* Due to the cache pressure, it is possible to fail when saving the operation. */
+                testutil_assert(ret == WT_ROLLBACK);
+                logger::log_msg(LOG_WARN,
+                  "The cache size reconfiguration could not be saved in the tracking table, ret: " +
+                    std::to_string(ret));
+                tc->transaction.rollback();
+            }
+            increase_cache = !increase_cache;
+        }
+    }
+
+    void
+    insert_operation(test_harness::thread_context *tc) override final
+    {
+        const uint64_t collection_count = tc->db.get_collection_count();
+        testutil_assert(collection_count > 0);
+        collection &coll = tc->db.get_collection(collection_count - 1);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+
+        while (tc->running()) {
+            tc->sleep();
+
+            /* Insert the current cache size value using a random key. */
+            const std::string key =
+              random_generator::instance().generate_pseudo_random_string(tc->key_size);
+            const uint64_t cache_size =
+              ((WT_CONNECTION_IMPL *)connection_manager::instance().get_connection())->cache_size;
+            /* Take into account the value size given in the test configuration file. */
+            const std::string value = tc->left_padding_to_string(cache_size, tc->value_size);
```

Comment:
This confused me at first glance as it looks like we're changing the size of `value` based on the current cache_size. Could we write a pseudorandom string instead?

Reply:
We are retrieving the cache size and then we format it using the `value_size`.
If the cache size was 1 and the `key_size` was 10, we would add padding of nine `0`s to the value `1`.